### PR TITLE
TST: Avoid bare pytest.raises in mult files

### PR DESCRIPTION
**Pull request by Vlek**
* [x]  ref #30999
 
* [x]  tests added / passed
 
* [x]  passes `black pandas`

* [x]  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/indexes/timedeltas/test_indexing.py [hunk 1]
```diff
@@ -184,7 +184,8 @@ def test_take_fill_value(self):
         with pytest.raises(ValueError, match=msg):
             idx.take(np.array([1, 0, -5]), fill_value=True)
 
-        with pytest.raises(IndexError):
+        msg = "index -5 is out of bounds for size 3"
```
Comment (jbrockmendel) [inline pandas/tests/indexes/timedeltas/test_indexing.py]: looks like this message isnt quite right?

Comment (Vlek) [reply pandas/tests/indexes/timedeltas/test_indexing.py]: Yes and no. There are OS differences. For Mac and I think Linux the message is `index -5 is out of bounds for **axis 0 with** size 3`. I've made the tests platform agnostic to look for that extra bit.

**Comment by Vlek** [comment]
I haven't the slightest clue why pandas-dev.pandas is taking so long while the others were successful. I would appreciate help in figuring out what's wrong.

**Comment by jreback** [comment]
thanks @Vlek

**Pull request merged**